retrofit/src/main/java/retrofit/DefaultCallAdapterFactory.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.concurrent.Executor;
+
+final class DefaultCallAdapterFactory implements CallAdapter.Factory {
+  private final Executor callbackExecutor;
```

Comment:
Required. The `NetworkCall` calls callbacks on the OkHttp callback thread. This bounces it onto the appropriate thread for the platform (e.g., main thread on Android).